lib/dialects/abstract/connection-manager.js

```diff
@@ -273,6 +274,7 @@ class ConnectionManager {
     return promise.then(() => {
       return this.pool.acquire(options.priority, options.type, options.useMaster)
         .then(mayBeConnection => this._determineConnection(mayBeConnection))
+        .catch(/TimeoutError/, err => { throw new errors.ConnectionAcquireTimeoutError(err); })
```

Comment:
Throw this error in `_determineConnection`, we also we want to cover cases of replication

Reply:
Maybe I miss your point, but when TimeoutError happens, process doesn't fall in _determineConnection?

Reply:
In case of timeout error `mayBeConnection` should be `TimeError` instance, which can be used to throw `ConnectionAcquireTimeoutError` inside `_determineConnection`

This is important because `_determineConnection` is used by both common pool and replication pool. Otherwise you will need to apply this same check to three different places (where `pool.acquire` is used)

Reply:
Timeout error is unfortunately exception from  case  described in #8330 - it is really rejected by pool, so _determineConnection is not called. By the way, is replication poll used directly, bypassing getConnection() method of connection-manager? Search didn't show me such cases